PROJECT_GUIDELINES.md

```diff
@@ -0,0 +1,163 @@
+
+#### Utils
+
+##### Unique IDs for components
+
+Many times it is necessary for components to have an `id="something"` attribute for things like `<label>` and various `aria-*` properties. To safely generate a unique id for a component, but to also allow a user supplied `id` attribute to work, follow the pattern using `guid` in our [calcite-value-list](https://github.com/Esri/calcite-app-components/blob/master/src/components/calcite-value-list/calcite-value-list.tsx).
+
+This will create a unique id attribute like `id="calcite-example-51af-0941-54ae-22c14d441beb"`, which should have a VERY low collision change since guid() generates IDs with `window.crypto.getRandomValues`. If a user supplies an `id`, it will respect the users `id`.
+
+##### i18n
+
+Components should require as a few text translations as possible. In general, this allows users to supply text values via slots and attributes and handle translations within their apps.
+
+If your component involves formatting numbers or dates use the [`Intl` APIs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl) for formating the display of numbers and dates in the component.
+
+To add RTL support to your components, use the internal `getElementDir` helper to apply the `dir` attribute to the component. That way, the `dir` attribute of the component will always match that of the document. See use of `getElementDir` in [calcite-panel](https://github.com/Esri/calcite-app-components/blob/master/src/components/calcite-panel/calcite-panel.tsx).
+
+Direction specific CSS can be implemented with CSS variables:
+
+```scss
+:host {
+  --calcite-tabs-tab-margin-start: 1.25rem;
+  --calcite-tabs-tab-margin-end: 0;
+}
+
+:host([dir="rtl"]) {
+  --calcite-tabs-tab-margin-start: 0;
+  --calcite-tabs-tab-margin-end: 1.25rem;
+}
+```
+
+- https://medium.com/stencil-tricks/implementing-internationalisation-i18n-with-stencil-5e6559554117 and https://codesandbox.io/s/43pmx55vo9
+- https://github.com/ionic-team/ionic-stencil-conference-app/issues/69
+
+#### Testing
+
+Components should have an automated test for any incoming features or bug fix. Make sure all tests pass as PRs will not be allowed to merge if there is a single test failure.
+
+We encourage writing expressive test cases and code that indicates intent. Use comments sparingly when the aforementioned can't be fully achieved. Keep it clean!
+
+Please see Stencil's doc for more info on [end-to-end](https://stenciljs.com/docs/end-to-end-testing) testing. See one of our test examples [here](https://github.com/Esri/calcite-app-components/blob/master/src/components/calcite-block/calcite-block.e2e.ts).
+
+#### Browser support
+
+See our [README.md](https://github.com/Esri/calcite-app-components/blob/master/README.md) for a list of supported browswers.
```

Comment:
Can you link to the browser support section? https://github.com/Esri/calcite-app-components/blob/master/README.md#browser-support

Reply:
Typo: `browswers` -> `browsers`